src/NUnitFramework/framework/Internal/Execution/TestWorker.cs

```diff
@@ -157,7 +175,18 @@ public void Start()
             _workerThread = new Thread(new ThreadStart(TestWorkerThreadProc));
             _workerThread.Name = Name;
 #if APARTMENT_STATE
-            _workerThread.SetApartmentState(WorkQueue.TargetApartment);
+#if NETSTANDARD2_0
+            if (OSPlatform.CurrentPlatform.IsWindows)
+            {
+#endif
+                _workerThread.SetApartmentState(WorkQueue.TargetApartment);
```

Comment:
Rather an OS detection, the .NET team intends folks to use feature detection—triggering and catching PlatformNotSuportedException, and maybe caching the fact that the feature is not available.

Reply:
~~Rather than OS detection, the .NET team intends folks to use feature detection—triggering and catching PlatformNotSupportedException, and maybe caching the fact that the feature is not available.~~

Well, I thought so, but https://blogs.msdn.microsoft.com/dotnet/2017/10/31/introducing-api-analyzer/#user-content-discovering-cross-platform-issues has `RuntimeInformation.IsOSPlatform(OSPlatform.Windows)`.